python/lsst/obs/base/cli/butler_cmd_test.py

```diff
@@ -29,36 +29,41 @@
 import click.testing
 
 from lsst.daf.butler.cli import butler
+from lsst.utils import doImport
 
 
 class ButlerCmdTestBase(metaclass=abc.ABCMeta):
     """Base class for tests of butler command line interface subcommands.
     Subclass from this, then `unittest.TestCase` to get a working test suite.
     """
 
-    @staticmethod
+    @property
     @abc.abstractmethod
-    def instrumentClass():
-        """Get the fully qualified instrument class.
+    def instrumentClassName(self):
+        """The fully qualified instrument class name.
 
         Returns
         -------
         `str`
-            The fully qualified instrument class.
+            The fully qualified instrument class name.
         """
         pass
 
-    @staticmethod
-    @abc.abstractmethod
-    def instrumentName():
-        """Get the instrument name.
+    @property
+    def instrument(self):
```

Comment:
`instrument` implies an instance of `Instrument` so I think `instrumentClass` would add clarity.